src/main/java/org/broadinstitute/hellbender/tools/funcotator/dataSources/DataSourceUtils.java

```diff
@@ -283,16 +360,52 @@ public static boolean isValidDirectory(final Path p) {
         return dataSourceFactories;
     }
 
+    private static FeatureInput<? extends Feature> createAndRegisterFeatureInputs(final Path dataSourceFile,
+                                                                                  final Properties dataSourceProperties,
+                                                                                  final GATKTool funcotatorToolInstance,
+                                                                                  final int lookaheadFeatureCachingInBp,
+                                                                                  final Class<? extends Feature> featureType) {
+        Utils.nonNull(dataSourceFile);
+        Utils.nonNull(dataSourceProperties);
+
+        final String name      = dataSourceProperties.getProperty(CONFIG_FILE_FIELD_NAME_NAME);
+        final String sourceFile = dataSourceFile.resolveSibling(dataSourceProperties.getProperty(CONFIG_FILE_FIELD_NAME_SRC_FILE)).toString();
+
+        // Get feature inputs by creating them with the funcotator tool instance itself:
+        return funcotatorToolInstance.addFeatureInputsAfterInitialization(sourceFile, name, featureType, lookaheadFeatureCachingInBp);
+    }
+
+    /**
+     * Create {@link FeatureInput<? extends Feature>} FOR TESTING ONLY.
+     * @param dataSourceFile
+     * @param dataSourceProperties
+     * @return
+     */
+    private static FeatureInput<? extends Feature> createFeatureInputsForTesting(final Path dataSourceFile,
+                                                                                 final Properties dataSourceProperties) {
+
+        Utils.nonNull(dataSourceFile);
+        Utils.nonNull(dataSourceProperties);
+
+        final String name      = dataSourceProperties.getProperty(CONFIG_FILE_FIELD_NAME_NAME);
+        final String sourceFile = dataSourceFile.resolveSibling(dataSourceProperties.getProperty(CONFIG_FILE_FIELD_NAME_SRC_FILE)).toString();
+
+        // Get feature inputs by creating them with the funcotator tool instance itself:
```

Comment:
Comment does not apply here.

Reply:
Yes it does.  This method both registers the `FeatureInput` objects with the `GATKTool` and also returns them.